channeldb/payment_control.go

```diff
@@ -615,19 +619,42 @@ func fetchPaymentBucketUpdate(tx kvdb.RwTx, paymentHash lntypes.Hash) (
 
 // nextPaymentSequence returns the next sequence number to store for a new
 // payment.
-func nextPaymentSequence(tx kvdb.RwTx) ([]byte, error) {
-	payments, err := tx.CreateTopLevelBucket(paymentsRootBucket)
-	if err != nil {
-		return nil, err
-	}
+func (p *PaymentControl) nextPaymentSequence() ([]byte, error) {
+	p.paymentSeqMx.Lock()
+	defer p.paymentSeqMx.Unlock()
+
+	// Set a new upper bound in the DB every 1000 payments to avoid
+	// conflicts on the sequence when using etcd.
+	if p.currPaymentSeq == p.storedPaymentSeq {
```

Comment:
nit: add to the comment here that this is also hit when we don't have anything cached yet, for example after a restart? Since both values will be zero then. Read through the code twice now and both times I thought "wait, what if we just started up but the sequence actually isn't 0 in the DB?". Only afterwards I saw that in that case both values are set with equal default values.

Reply:
Sorry about the confusion, indeed it was a bit implicit without a proper comment. :sweat_smile:  Added, ptal.